--- build_defs/internal_shell.bzl
+++ build_defs/internal_shell.bzl
@@ -32,7 +32,7 @@
     native.genrule(
         name = name + "_genrule",
         srcs = srcs,
-        exec_tools = tools,
+        tools = tools,
         outs = [name + ".sh"],
         cmd = "cat <<'EOF' >$(OUTS)\n#!/bin/bash -exu\n%s\nEOF\n" % cmd,
         visibility = ["//visibility:private"],
@@ -77,7 +77,7 @@
     native.genrule(
         name = name + "_genrule",
         srcs = srcs,
-        exec_tools = tools,
+        tools = tools,
         outs = [name + ".sh"],
         cmd = "cat <<'EOF' >$(OUTS)\n#!/bin/bash -exu\n%s\nEOF\n" % cmd,
         visibility = ["//visibility:private"],

--- src/google/protobuf/BUILD.bazel
+++ src/google/protobuf/BUILD.bazel
@@ -138,7 +138,7 @@
             --proto_path=$$(dirname $$(dirname $$(dirname $(location any.proto)))) \
             $(SRCS)
     """,
-    exec_tools = ["//:protoc"],
+    tools = ["//:protoc"],
     visibility = ["//visibility:private"],
 )
 
