diff --git bazel/upb_proto_library.bzl bazel/upb_proto_library.bzl
index d444ae20..45dc47b5 100644
--- bazel/upb_proto_library.bzl
+++ bazel/upb_proto_library.bzl
@@ -123,6 +123,12 @@ def _cc_library_func(ctx, name, hdrs, srcs, copts, dep_ccinfos):
         cc_toolchain = toolchain,
         compilation_outputs = compilation_outputs,
         linking_contexts = linking_contexts,
+        # On Windows, we can’t build dynamic libraries because they would need
+        # to export all symbols to be usable.
+        disallow_dynamic_library = cc_common.is_enabled(
+            feature_configuration = feature_configuration,
+            feature_name = "targets_windows",
+        ),
         **blaze_only_args
     )
 
